src/Umbraco.Deploy.Contrib/Migrators/ReplaceMatryoshkaArtifactMigrator.cs

```diff
@@ -0,0 +1,100 @@
+using System.Linq;
+using System;
+using Umbraco.Cms.Core.Deploy;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Semver;
+using Umbraco.Deploy.Core.Migrators;
+using Umbraco.Deploy.Infrastructure.Artifacts;
+using Umbraco.Deploy.Infrastructure.Artifacts.ContentType;
+using System.Collections.Generic;
+using Umbraco.Extensions;
+
+namespace Umbraco.Deploy.Contrib.Migrators;
+
+/// <summary>
+/// Migrates the <see cref="DataTypeArtifact" /> to remove Matryoshka data types (using the <see cref="EditorAlias" /> editor) and
+/// <see cref="ContentTypeArtifactBase" /> to replace properties using the removed Matryoshka data types with native groups.
+/// </summary>
+public sealed class ReplaceMatryoshkaArtifactMigrator : ArtifactMigratorBase<IArtifact>
+{
+    private const string EditorAlias = "Our.Umbraco.Matryoshka.GroupSeparator";
+
+    private readonly HashSet<Guid> _removedDataTypeKeys = [];
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ReplaceMatryoshkaArtifactMigrator" /> class.
+    /// </summary>
+    public ReplaceMatryoshkaArtifactMigrator()
+        // Matryoshka was only supported on v8
+        => MaxVersion = new SemVersion(9, 0, 0);
```

Comment:
We could also set a minimum version like this (note that the versions are Deploy versions, so CMS v8 is Deploy v4):
```suggestion
    {
        // Matryoshka was only supported on v8
        MinVersion = new SemVersion(4, 0, 0);
        MaxVersion = new SemVersion(9, 0, 0);
    }
```

I've omitted this, because you'll most likely only add this migrator if you want to import from v8 anyway...